src/hooks/useForm.js

```diff
@@ -0,0 +1,83 @@
+import { useState, useCallback } from 'react';
+
+const useForm = (
+  {
+    initialValues,
+    onSubmit,
+    validator = () => {},
+    validateOnChange = false,
+    validateOnBlur = false,
+  },
+  ...dependencies
+) => {
+  const [values, setValues] = useState(initialValues);
+  const [errors, setErrors] = useState({});
+  const [blurredFields, setBlurredFields] = useState({});
+
+  const handleSubmit = useCallback(() => {
+    const newErrors = validator(values) || {};
+    const valid = !Object.values(newErrors)
+      .filter(error => !!error)
+      .reduce((acc, error) => {
+        error.forEach(e => acc.push(e));
+        return acc;
+      }, []).length;
+    if (valid) {
+      onSubmit(values);
+    } else {
+      setErrors(newErrors);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [values, setErrors, validator, onSubmit, ...dependencies]);
+
+  const runValidations = useCallback(
```

Comment:
not sure if this should be part of the hook, could be on a separate file? i'm not sure if it's aligned with the other responsibilities, wdyt?

Reply:
The runValidations (as the name suggests) is in charge of just running the validator function that is passed to the hook, so the actual validation lives outside of the hook.